Error on mac: Unable to change file mode on `mnt//res/fonts/CJKBiángRegular36.font`: No such file or directory

I have tried to run `./Meta/serenity.sh run`, this is the output:
```
build_jakt: /Users/ljian/CTF/Kernel/serenity/Toolchain/Local/jakt
[dependencies] Checking whether 'ninja' is available...
[dependencies] Checking whether 'cmake' is available...
[dependencies] Checking whether your C++ compiler works...
~/CTF/Kernel/serenity/Toolchain/Tarballs ~/CTF/Kernel/serenity/Toolchain
Using existing Jakt source directory
~/CTF/Kernel/serenity/Toolchain
Expected 1fed928d0abf08188e48fe765ab68e5047c05ec2 but found none (for local host)
~/CTF/Kernel/serenity/Toolchain/Build/jakt ~/CTF/Kernel/serenity/Toolchain
XXX configure jakt
[jakt/configure] -- The CXX compiler identification is AppleClang 17.0.0.17000013
[jakt/configure] -- The C compiler identification is AppleClang 17.0.0.17000013
[jakt/configure] -- Detecting CXX compiler ABI info
[jakt/configure] -- Detecting CXX compiler ABI info - done
[jakt/configure] -- Check for working CXX compiler: /usr/bin/c++ - skipped
[jakt/configure] -- Detecting CXX compile features
[jakt/configure] -- Detecting CXX compile features - done
[jakt/configure] -- Detecting C compiler ABI info
[jakt/configure] -- Detecting C compiler ABI info - done
[jakt/configure] -- Check for working C compiler: /usr/bin/cc - skipped
[jakt/configure] -- Detecting C compile features
[jakt/configure] -- Detecting C compile features - done
[jakt/configure] -- Performing Test CMAKE_HAVE_LIBC_PTHREAD
[jakt/configure] -- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
[jakt/configure] -- Found Threads: TRUE
[jakt/configure] -- Could NOT find Clang (missing: Clang_DIR)
[jakt/configure] -- Using none as the cpp import processor (using Clang=OFF, resource dir=)
[jakt/configure] -- Using preexisting AK directory from SERENITY_SOURCE_DIR: /Users/ljian/CTF/Kernel/serenity/Toolchain/../AK
[jakt/configure] CMake Warning (dev) at /opt/homebrew/share/cmake/Modules/FetchContent.cmake:1953 (message):
[jakt/configure]   Calling FetchContent_Populate(Serenity) is deprecated, call
[jakt/configure]   FetchContent_MakeAvailable(Serenity) instead.  Policy CMP0169 can be set to
[jakt/configure]   OLD to allow FetchContent_Populate(Serenity) to be called directly for now,
[jakt/configure]   but the ability to call it with declared details will be removed completely
[jakt/configure]   in a future version.
[jakt/configure] Call Stack (most recent call first):
[jakt/configure]   runtime/CMakeLists.txt:24 (FetchContent_Populate)
[jakt/configure] This warning is for project developers.  Use -Wno-dev to suppress it.
[jakt/configure] 
[jakt/configure] -- Making the jakt runtime available for arm64-unknown-darwin-unknown
[jakt/configure] -- If this is windows, '' should be nonempty
[jakt/configure] -- Building for target arm64-unknown-darwin-unknown
[jakt/configure] CMake Warning (dev) at cmake/symlink.cmake:41 (add_custom_command):
[jakt/configure]   The following keywords are not supported when using
[jakt/configure]   add_custom_command(TARGET): DEPENDS.
[jakt/configure] 
[jakt/configure]   Policy CMP0175 is not set: add_custom_command() rejects invalid arguments.
[jakt/configure]   Run "cmake --help-policy CMP0175" for policy details.  Use the cmake_policy
[jakt/configure]   command to set the policy and suppress this warning.
[jakt/configure] Call Stack (most recent call first):
[jakt/configure]   CMakeLists.txt:294 (add_symlink_command)
[jakt/configure] This warning is for project developers.  Use -Wno-dev to suppress it.
[jakt/configure] 
[jakt/configure] -- Configuring done (1.8s)
[jakt/configure] -- Generating done (0.0s)
[jakt/configure] -- Build files have been written to: /Users/ljian/CTF/Kernel/serenity/Toolchain/Build/jakt
XXX build jakt
[jakt/build] [0/2] Re-checking globbed directories...
[jakt/build] [238/240] Building CXX object CMakeFiles/jakt_stage1.dir/jakt_stage1_tmp/types.cpp.o
FAILED: CMakeFiles/jakt_stage1.dir/jakt_stage1_tmp/types.cpp.o 
/opt/homebrew/bin/ccache /usr/bin/c++  -I/Users/ljian/CTF/Kernel/serenity/Toolchain/Tarballs/jakt-1fed928d0abf08188e48fe765ab68e5047c05ec2/runtime -O3 -DNDEBUG -std=gnu++20 -arch arm64 -Wall -Wextra -Werror -Wno-maybe-uninitialized -Wno-unused-local-typedefs -Wno-unused-function -Wno-unused-variable -Wno-unused-parameter -Wno-unused-but-set-variable -Wno-unused-result -Wno-implicit-fallthrough -Wno-trigraphs -Wno-parentheses-equality -Wno-unqualified-std-cast-call -Wno-user-defined-literals -Wno-literal-suffix -Wno-return-type -Wno-deprecated-declarations -Wno-unknown-warning-option -Wno-unused-command-line-argument -Wno-unused-lambda-capture -Wno-reorder-ctor -Wno-dangling-reference -Wno-type-limits -Wno-unknown-attributes -Wno-invalid-offsetof -fno-exceptions -fdiagnostics-color=always -MD -MT CMakeFiles/jakt_stage1.dir/jakt_stage1_tmp/types.cpp.o -MF CMakeFiles/jakt_stage1.dir/jakt_stage1_tmp/types.cpp.o.d -o CMakeFiles/jakt_stage1.dir/jakt_stage1_tmp/types.cpp.o -c /Users/ljian/CTF/Kernel/serenity/Toolchain/Build/jakt/jakt_stage1_tmp/types.cpp
/Users/ljian/CTF/Kernel/serenity/Toolchain/Build/jakt/jakt_stage1_tmp/types.cpp:2735:1: error: ignoring temporary created by a constructor declared with 'nodiscard' attribute [-Werror,-Wunused-value]
 2735 | TRY((this->template for_each_scope_accessible_unqualified_from_scope<JaktInternal::Tuple<JaktInternal::DynamicArray<Jakt::ids::FunctionId>,Jakt::ids::ScopeId>>(parent_scope_id,[function_name, &result_ids, &result_scope, this](Jakt::types::ResolutionMixin mixin, JaktInternal::Optional<ByteString> name_override, bool is_alias) -> ErrorOr<typename Jakt::utility::IterationDecision<JaktInternal::Tuple<JaktInternal::DynamicArray<Jakt::ids::FunctionId>,Jakt::ids::ScopeId>>> {{
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 2736 | if (!mixin.mixin_functions){
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 2737 | return Jakt::utility::IterationDecision<JaktInternal::Tuple<JaktInternal::DynamicArray<Jakt::ids::FunctionId>,Jakt::ids::ScopeId>>::Continue();
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 2738 | }
      | ~
 2739 | NonnullRefPtr<Jakt::types::Scope> const scope = this->get_scope(mixin.scope_id);
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 2740 | JaktInternal::Optional<JaktInternal::DynamicArray<Jakt::ids::FunctionId>> const maybe_functions = scope->functions.get(function_name);
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 2741 | if (maybe_functions.has_value()){
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 2742 | result_ids.push_values(maybe_functions.value());
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 2743 | if (!result_scope.has_value()){
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 2744 | result_scope = mixin.scope_id;
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 2745 | }
      | ~
 2746 | }
      | ~
 2747 | return Jakt::utility::IterationDecision<JaktInternal::Tuple<JaktInternal::DynamicArray<Jakt::ids::FunctionId>,Jakt::ids::ScopeId>>::Continue();
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 2748 | }
      | ~
 2749 | }
      | ~
 2750 | ,ignore_mixin_scopes,root_scope)));
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/ljian/CTF/Kernel/serenity/Toolchain/Tarballs/jakt-1fed928d0abf08188e48fe765ab68e5047c05ec2/runtime/AK/Try.h:33:9: note: expanded from macro 'TRY'
   33 |         _temporary_result.release_value();                                                           \
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
[jakt/build] [239/240] Building CXX object CMakeFiles/jakt_stage1.dir/jakt_stage1_tmp/typechecker.cpp.o
FAILED: CMakeFiles/jakt_stage1.dir/jakt_stage1_tmp/typechecker.cpp.o 
/opt/homebrew/bin/ccache /usr/bin/c++  -I/Users/ljian/CTF/Kernel/serenity/Toolchain/Tarballs/jakt-1fed928d0abf08188e48fe765ab68e5047c05ec2/runtime -O3 -DNDEBUG -std=gnu++20 -arch arm64 -Wall -Wextra -Werror -Wno-maybe-uninitialized -Wno-unused-local-typedefs -Wno-unused-function -Wno-unused-variable -Wno-unused-parameter -Wno-unused-but-set-variable -Wno-unused-result -Wno-implicit-fallthrough -Wno-trigraphs -Wno-parentheses-equality -Wno-unqualified-std-cast-call -Wno-user-defined-literals -Wno-literal-suffix -Wno-return-type -Wno-deprecated-declarations -Wno-unknown-warning-option -Wno-unused-command-line-argument -Wno-unused-lambda-capture -Wno-reorder-ctor -Wno-dangling-reference -Wno-type-limits -Wno-unknown-attributes -Wno-invalid-offsetof -fno-exceptions -fdiagnostics-color=always -MD -MT CMakeFiles/jakt_stage1.dir/jakt_stage1_tmp/typechecker.cpp.o -MF CMakeFiles/jakt_stage1.dir/jakt_stage1_tmp/typechecker.cpp.o.d -o CMakeFiles/jakt_stage1.dir/jakt_stage1_tmp/typechecker.cpp.o -c /Users/ljian/CTF/Kernel/serenity/Toolchain/Build/jakt/jakt_stage1_tmp/typechecker.cpp
/Users/ljian/CTF/Kernel/serenity/Toolchain/Build/jakt/jakt_stage1_tmp/typechecker.cpp:9423:1: error: ignoring temporary created by a constructor declared with 'nodiscard' attribute [-Werror,-Wunused-value]
 9423 | TRY((this->typecheck_statement(var_decl_stmt,wrapped_scope_id,Jakt::types::SafetyMode::Safe(),JaktInternal::OptionalNone())));
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/ljian/CTF/Kernel/serenity/Toolchain/Tarballs/jakt-1fed928d0abf08188e48fe765ab68e5047c05ec2/runtime/AK/Try.h:33:9: note: expanded from macro 'TRY'
   33 |         _temporary_result.release_value();                                                           \
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
ninja: build stopped: subcommand failed.
build_jakt: /Users/ljian/CTF/Kernel/serenity/Toolchain/Local/jakt
[dependencies] Checking whether 'ninja' is available...
[dependencies] Checking whether 'cmake' is available...
[dependencies] Checking whether your C++ compiler works...
~/CTF/Kernel/serenity/Toolchain/Tarballs ~/CTF/Kernel/serenity/Toolchain
Using existing Jakt source directory
~/CTF/Kernel/serenity/Toolchain
Expected 1fed928d0abf08188e48fe765ab68e5047c05ec2 but found none (for local host)
~/CTF/Kernel/serenity/Toolchain/Build/jakt ~/CTF/Kernel/serenity/Toolchain
XXX configure jakt
[jakt/configure] -- The CXX compiler identification is AppleClang 17.0.0.17000013
[jakt/configure] -- The C compiler identification is AppleClang 17.0.0.17000013
[jakt/configure] -- Detecting CXX compiler ABI info
[jakt/configure] -- Detecting CXX compiler ABI info - done
[jakt/configure] -- Check for working CXX compiler: /usr/bin/c++ - skipped
[jakt/configure] -- Detecting CXX compile features
[jakt/configure] -- Detecting CXX compile features - done
[jakt/configure] -- Detecting C compiler ABI info
[jakt/configure] -- Detecting C compiler ABI info - done
[jakt/configure] -- Check for working C compiler: /usr/bin/cc - skipped
[jakt/configure] -- Detecting C compile features
[jakt/configure] -- Detecting C compile features - done
[jakt/configure] -- Performing Test CMAKE_HAVE_LIBC_PTHREAD
[jakt/configure] -- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
[jakt/configure] -- Found Threads: TRUE
[jakt/configure] -- Could NOT find Clang (missing: Clang_DIR)
[jakt/configure] -- Using none as the cpp import processor (using Clang=OFF, resource dir=)
[jakt/configure] -- Using preexisting AK directory from SERENITY_SOURCE_DIR: /Users/ljian/CTF/Kernel/serenity/Toolchain/../AK
[jakt/configure] CMake Warning (dev) at /opt/homebrew/share/cmake/Modules/FetchContent.cmake:1953 (message):
[jakt/configure]   Calling FetchContent_Populate(Serenity) is deprecated, call
[jakt/configure]   FetchContent_MakeAvailable(Serenity) instead.  Policy CMP0169 can be set to
[jakt/configure]   OLD to allow FetchContent_Populate(Serenity) to be called directly for now,
[jakt/configure]   but the ability to call it with declared details will be removed completely
[jakt/configure]   in a future version.
[jakt/configure] Call Stack (most recent call first):
[jakt/configure]   runtime/CMakeLists.txt:24 (FetchContent_Populate)
[jakt/configure] This warning is for project developers.  Use -Wno-dev to suppress it.
[jakt/configure] 
[jakt/configure] -- Making the jakt runtime available for arm64-unknown-darwin-unknown
[jakt/configure] -- If this is windows, '' should be nonempty
[jakt/configure] -- Building for target arm64-unknown-darwin-unknown
[jakt/configure] CMake Warning (dev) at cmake/symlink.cmake:41 (add_custom_command):
[jakt/configure]   The following keywords are not supported when using
[jakt/configure]   add_custom_command(TARGET): DEPENDS.
[jakt/configure] 
[jakt/configure]   Policy CMP0175 is not set: add_custom_command() rejects invalid arguments.
[jakt/configure]   Run "cmake --help-policy CMP0175" for policy details.  Use the cmake_policy
[jakt/configure]   command to set the policy and suppress this warning.
[jakt/configure] Call Stack (most recent call first):
[jakt/configure]   CMakeLists.txt:294 (add_symlink_command)
[jakt/configure] This warning is for project developers.  Use -Wno-dev to suppress it.
[jakt/configure] 
[jakt/configure] -- Configuring done (1.8s)
[jakt/configure] -- Generating done (0.0s)
[jakt/configure] -- Build files have been written to: /Users/ljian/CTF/Kernel/serenity/Toolchain/Build/jakt
XXX build jakt
[jakt/build] [0/2] Re-checking globbed directories...
[jakt/build] [238/240] Building CXX object CMakeFiles/jakt_stage1.dir/jakt_stage1_tmp/types.cpp.o
FAILED: CMakeFiles/jakt_stage1.dir/jakt_stage1_tmp/types.cpp.o 
/opt/homebrew/bin/ccache /usr/bin/c++  -I/Users/ljian/CTF/Kernel/serenity/Toolchain/Tarballs/jakt-1fed928d0abf08188e48fe765ab68e5047c05ec2/runtime -O3 -DNDEBUG -std=gnu++20 -arch arm64 -Wall -Wextra -Werror -Wno-maybe-uninitialized -Wno-unused-local-typedefs -Wno-unused-function -Wno-unused-variable -Wno-unused-parameter -Wno-unused-but-set-variable -Wno-unused-result -Wno-implicit-fallthrough -Wno-trigraphs -Wno-parentheses-equality -Wno-unqualified-std-cast-call -Wno-user-defined-literals -Wno-literal-suffix -Wno-return-type -Wno-deprecated-declarations -Wno-unknown-warning-option -Wno-unused-command-line-argument -Wno-unused-lambda-capture -Wno-reorder-ctor -Wno-dangling-reference -Wno-type-limits -Wno-unknown-attributes -Wno-invalid-offsetof -fno-exceptions -fdiagnostics-color=always -MD -MT CMakeFiles/jakt_stage1.dir/jakt_stage1_tmp/types.cpp.o -MF CMakeFiles/jakt_stage1.dir/jakt_stage1_tmp/types.cpp.o.d -o CMakeFiles/jakt_stage1.dir/jakt_stage1_tmp/types.cpp.o -c /Users/ljian/CTF/Kernel/serenity/Toolchain/Build/jakt/jakt_stage1_tmp/types.cpp
/Users/ljian/CTF/Kernel/serenity/Toolchain/Build/jakt/jakt_stage1_tmp/types.cpp:2735:1: error: ignoring temporary created by a constructor declared with 'nodiscard' attribute [-Werror,-Wunused-value]
 2735 | TRY((this->template for_each_scope_accessible_unqualified_from_scope<JaktInternal::Tuple<JaktInternal::DynamicArray<Jakt::ids::FunctionId>,Jakt::ids::ScopeId>>(parent_scope_id,[function_name, &result_ids, &result_scope, this](Jakt::types::ResolutionMixin mixin, JaktInternal::Optional<ByteString> name_override, bool is_alias) -> ErrorOr<typename Jakt::utility::IterationDecision<JaktInternal::Tuple<JaktInternal::DynamicArray<Jakt::ids::FunctionId>,Jakt::ids::ScopeId>>> {{
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 2736 | if (!mixin.mixin_functions){
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 2737 | return Jakt::utility::IterationDecision<JaktInternal::Tuple<JaktInternal::DynamicArray<Jakt::ids::FunctionId>,Jakt::ids::ScopeId>>::Continue();
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 2738 | }
      | ~
 2739 | NonnullRefPtr<Jakt::types::Scope> const scope = this->get_scope(mixin.scope_id);
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 2740 | JaktInternal::Optional<JaktInternal::DynamicArray<Jakt::ids::FunctionId>> const maybe_functions = scope->functions.get(function_name);
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 2741 | if (maybe_functions.has_value()){
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 2742 | result_ids.push_values(maybe_functions.value());
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 2743 | if (!result_scope.has_value()){
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 2744 | result_scope = mixin.scope_id;
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 2745 | }
      | ~
 2746 | }
      | ~
 2747 | return Jakt::utility::IterationDecision<JaktInternal::Tuple<JaktInternal::DynamicArray<Jakt::ids::FunctionId>,Jakt::ids::ScopeId>>::Continue();
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 2748 | }
      | ~
 2749 | }
      | ~
 2750 | ,ignore_mixin_scopes,root_scope)));
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/ljian/CTF/Kernel/serenity/Toolchain/Tarballs/jakt-1fed928d0abf08188e48fe765ab68e5047c05ec2/runtime/AK/Try.h:33:9: note: expanded from macro 'TRY'
   33 |         _temporary_result.release_value();                                                           \
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
[jakt/build] [239/240] Building CXX object CMakeFiles/jakt_stage1.dir/jakt_stage1_tmp/typechecker.cpp.o
FAILED: CMakeFiles/jakt_stage1.dir/jakt_stage1_tmp/typechecker.cpp.o 
/opt/homebrew/bin/ccache /usr/bin/c++  -I/Users/ljian/CTF/Kernel/serenity/Toolchain/Tarballs/jakt-1fed928d0abf08188e48fe765ab68e5047c05ec2/runtime -O3 -DNDEBUG -std=gnu++20 -arch arm64 -Wall -Wextra -Werror -Wno-maybe-uninitialized -Wno-unused-local-typedefs -Wno-unused-function -Wno-unused-variable -Wno-unused-parameter -Wno-unused-but-set-variable -Wno-unused-result -Wno-implicit-fallthrough -Wno-trigraphs -Wno-parentheses-equality -Wno-unqualified-std-cast-call -Wno-user-defined-literals -Wno-literal-suffix -Wno-return-type -Wno-deprecated-declarations -Wno-unknown-warning-option -Wno-unused-command-line-argument -Wno-unused-lambda-capture -Wno-reorder-ctor -Wno-dangling-reference -Wno-type-limits -Wno-unknown-attributes -Wno-invalid-offsetof -fno-exceptions -fdiagnostics-color=always -MD -MT CMakeFiles/jakt_stage1.dir/jakt_stage1_tmp/typechecker.cpp.o -MF CMakeFiles/jakt_stage1.dir/jakt_stage1_tmp/typechecker.cpp.o.d -o CMakeFiles/jakt_stage1.dir/jakt_stage1_tmp/typechecker.cpp.o -c /Users/ljian/CTF/Kernel/serenity/Toolchain/Build/jakt/jakt_stage1_tmp/typechecker.cpp
/Users/ljian/CTF/Kernel/serenity/Toolchain/Build/jakt/jakt_stage1_tmp/typechecker.cpp:9423:1: error: ignoring temporary created by a constructor declared with 'nodiscard' attribute [-Werror,-Wunused-value]
 9423 | TRY((this->typecheck_statement(var_decl_stmt,wrapped_scope_id,Jakt::types::SafetyMode::Safe(),JaktInternal::OptionalNone())));
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/ljian/CTF/Kernel/serenity/Toolchain/Tarballs/jakt-1fed928d0abf08188e48fe765ab68e5047c05ec2/runtime/AK/Try.h:33:9: note: expanded from macro 'TRY'
   33 |         _temporary_result.release_value();                                                           \
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
ninja: build stopped: subcommand failed.
[0/5] Performing build step for 'lagom'
ninja: no work to do.
[1/5] Performing install step for 'lagom'
[0/1] Install the project...
-- Install configuration: ""
[2/5] Performing build step for 'serenity'
[1/1] Running utility command for create_compat_ssp_libraries
[3/5] Performing install step for 'serenity'
[1/2] Install the project...
-- Install configuration: ""
[5/5] Completed 'serenity'
ninja: Entering directory `/Users/ljian/CTF/Kernel/serenity/Build/aarch64'
[1/2] Install the project...
-- Install configuration: ""
ninja: Entering directory `/Users/ljian/CTF/Kernel/serenity/Build/aarch64'
[0/1] cd /Users/ljian/CTF/Kernel/serenity/Build/aarch64 &&... /Users/ljian/CTF/Kernel/serenity/Meta/build-image-qemu.sh
Password:
checking existing image
e2fsck 1.47.2 (1-Jan-2025)
Pass 1: Checking inodes, blocks, and sizes
Pass 2: Checking directory structure
Pass 3: Checking directory connectivity
Pass 4: Checking reference counts
Pass 5: Checking group summary information
Free blocks count wrong for group #0 (15397, counted=15387).
Fix? yes

Free blocks count wrong for group #6 (0, counted=14).
Fix? yes

Free blocks count wrong for group #7 (1056, counted=1098).
Fix? yes

Free blocks count wrong (274308, counted=274354).
Fix? yes

Free inodes count wrong for group #4 (5741, counted=5756).
Fix? yes

Free inodes count wrong (96909, counted=96924).
Fix? yes


_disk_image: ***** FILE SYSTEM WAS MODIFIED *****
_disk_image: 35684/132608 files (0.2% non-contiguous), 249934/524288 blocks
done
mounting filesystem... mount_macfuse: mount point /Users/ljian/CTF/Kernel/serenity/Build/aarch64/mnt is itself on a macFUSE volume
die: could not mount filesystem and genext2fs is missing
FAILED: CMakeFiles/qemu-image _disk_image /Users/ljian/CTF/Kernel/serenity/Build/aarch64/CMakeFiles/qemu-image /Users/ljian/CTF/Kernel/serenity/Build/aarch64/_disk_image 
cd /Users/ljian/CTF/Kernel/serenity/Build/aarch64 && /opt/homebrew/bin/cmake -E env SERENITY_SOURCE_DIR=/Users/ljian/CTF/Kernel/serenity SERENITY_ARCH=aarch64 SERENITY_TOOLCHAIN=GNU LLVM_VERSION=13.2.0 /Users/ljian/CTF/Kernel/serenity/Meta/build-image-qemu.sh
ninja: build stopped: subcommand failed.
~/C/K/serenity (master|✚1) [1]$ ./Meta/serenity.sh run^C                                                         (base) 
~/C/K/serenity (master|✚1) [1]$ uname -a                                                                         (base) 
Darwin RocketMacAir.local 24.3.0 Darwin Kernel Version 24.3.0: Thu Jan  2 20:23:36 PST 2025; root:xnu-11215.81.4~3/RELEASE_ARM64_T8112 arm64
~/C/K/serenity (master|✚1) $ brew install genext2fs                                                              (base) 
==> Downloading https://formulae.brew.sh/api/formula.jws.json
==> Downloading https://formulae.brew.sh/api/cask.jws.json
==> Downloading https://ghcr.io/v2/homebrew/core/genext2fs/manifests/1.5.0
################################################################################################################# 100.0%
==> Fetching genext2fs
==> Downloading https://ghcr.io/v2/homebrew/core/genext2fs/blobs/sha256:cc9fdd7364d47e9a525a63f18f8556678f95c84785aafa58
################################################################################################################# 100.0%
==> Pouring genext2fs--1.5.0.arm64_sequoia.bottle.tar.gz
🍺  /opt/homebrew/Cellar/genext2fs/1.5.0: 10 files, 121.3KB
==> Running `brew cleanup genext2fs`...
Disable this behaviour by setting HOMEBREW_NO_INSTALL_CLEANUP.
Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
~/C/K/serenity (master|✚1) $ ./Meta/serenity.sh run                                                              (base) 
build_jakt: /Users/ljian/CTF/Kernel/serenity/Toolchain/Local/jakt
[dependencies] Checking whether 'ninja' is available...
[dependencies] Checking whether 'cmake' is available...
[dependencies] Checking whether your C++ compiler works...
~/CTF/Kernel/serenity/Toolchain/Tarballs ~/CTF/Kernel/serenity/Toolchain
Using existing Jakt source directory
~/CTF/Kernel/serenity/Toolchain
Expected 1fed928d0abf08188e48fe765ab68e5047c05ec2 but found none (for local host)
~/CTF/Kernel/serenity/Toolchain/Build/jakt ~/CTF/Kernel/serenity/Toolchain
XXX configure jakt
[jakt/configure] -- The CXX compiler identification is AppleClang 17.0.0.17000013
[jakt/configure] -- The C compiler identification is AppleClang 17.0.0.17000013
[jakt/configure] -- Detecting CXX compiler ABI info
[jakt/configure] -- Detecting CXX compiler ABI info - done
[jakt/configure] -- Check for working CXX compiler: /usr/bin/c++ - skipped
[jakt/configure] -- Detecting CXX compile features
[jakt/configure] -- Detecting CXX compile features - done
[jakt/configure] -- Detecting C compiler ABI info
[jakt/configure] -- Detecting C compiler ABI info - done
[jakt/configure] -- Check for working C compiler: /usr/bin/cc - skipped
[jakt/configure] -- Detecting C compile features
[jakt/configure] -- Detecting C compile features - done
[jakt/configure] -- Performing Test CMAKE_HAVE_LIBC_PTHREAD
[jakt/configure] -- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
[jakt/configure] -- Found Threads: TRUE
[jakt/configure] -- Could NOT find Clang (missing: Clang_DIR)
[jakt/configure] -- Using none as the cpp import processor (using Clang=OFF, resource dir=)
[jakt/configure] -- Using preexisting AK directory from SERENITY_SOURCE_DIR: /Users/ljian/CTF/Kernel/serenity/Toolchain/../AK
[jakt/configure] CMake Warning (dev) at /opt/homebrew/share/cmake/Modules/FetchContent.cmake:1953 (message):
[jakt/configure]   Calling FetchContent_Populate(Serenity) is deprecated, call
[jakt/configure]   FetchContent_MakeAvailable(Serenity) instead.  Policy CMP0169 can be set to
[jakt/configure]   OLD to allow FetchContent_Populate(Serenity) to be called directly for now,
[jakt/configure]   but the ability to call it with declared details will be removed completely
[jakt/configure]   in a future version.
[jakt/configure] Call Stack (most recent call first):
[jakt/configure]   runtime/CMakeLists.txt:24 (FetchContent_Populate)
[jakt/configure] This warning is for project developers.  Use -Wno-dev to suppress it.
[jakt/configure] 
[jakt/configure] -- Making the jakt runtime available for arm64-unknown-darwin-unknown
[jakt/configure] -- If this is windows, '' should be nonempty
[jakt/configure] -- Building for target arm64-unknown-darwin-unknown
[jakt/configure] CMake Warning (dev) at cmake/symlink.cmake:41 (add_custom_command):
[jakt/configure]   The following keywords are not supported when using
[jakt/configure]   add_custom_command(TARGET): DEPENDS.
[jakt/configure] 
[jakt/configure]   Policy CMP0175 is not set: add_custom_command() rejects invalid arguments.
[jakt/configure]   Run "cmake --help-policy CMP0175" for policy details.  Use the cmake_policy
[jakt/configure]   command to set the policy and suppress this warning.
[jakt/configure] Call Stack (most recent call first):
[jakt/configure]   CMakeLists.txt:294 (add_symlink_command)
[jakt/configure] This warning is for project developers.  Use -Wno-dev to suppress it.
[jakt/configure] 
[jakt/configure] -- Configuring done (1.9s)
[jakt/configure] -- Generating done (0.0s)
[jakt/configure] -- Build files have been written to: /Users/ljian/CTF/Kernel/serenity/Toolchain/Build/jakt
XXX build jakt
[jakt/build] [0/2] Re-checking globbed directories...
[jakt/build] [238/240] Building CXX object CMakeFiles/jakt_stage1.dir/jakt_stage1_tmp/types.cpp.o
FAILED: CMakeFiles/jakt_stage1.dir/jakt_stage1_tmp/types.cpp.o 
/opt/homebrew/bin/ccache /usr/bin/c++  -I/Users/ljian/CTF/Kernel/serenity/Toolchain/Tarballs/jakt-1fed928d0abf08188e48fe765ab68e5047c05ec2/runtime -O3 -DNDEBUG -std=gnu++20 -arch arm64 -Wall -Wextra -Werror -Wno-maybe-uninitialized -Wno-unused-local-typedefs -Wno-unused-function -Wno-unused-variable -Wno-unused-parameter -Wno-unused-but-set-variable -Wno-unused-result -Wno-implicit-fallthrough -Wno-trigraphs -Wno-parentheses-equality -Wno-unqualified-std-cast-call -Wno-user-defined-literals -Wno-literal-suffix -Wno-return-type -Wno-deprecated-declarations -Wno-unknown-warning-option -Wno-unused-command-line-argument -Wno-unused-lambda-capture -Wno-reorder-ctor -Wno-dangling-reference -Wno-type-limits -Wno-unknown-attributes -Wno-invalid-offsetof -fno-exceptions -fdiagnostics-color=always -MD -MT CMakeFiles/jakt_stage1.dir/jakt_stage1_tmp/types.cpp.o -MF CMakeFiles/jakt_stage1.dir/jakt_stage1_tmp/types.cpp.o.d -o CMakeFiles/jakt_stage1.dir/jakt_stage1_tmp/types.cpp.o -c /Users/ljian/CTF/Kernel/serenity/Toolchain/Build/jakt/jakt_stage1_tmp/types.cpp
/Users/ljian/CTF/Kernel/serenity/Toolchain/Build/jakt/jakt_stage1_tmp/types.cpp:2735:1: error: ignoring temporary created by a constructor declared with 'nodiscard' attribute [-Werror,-Wunused-value]
 2735 | TRY((this->template for_each_scope_accessible_unqualified_from_scope<JaktInternal::Tuple<JaktInternal::DynamicArray<Jakt::ids::FunctionId>,Jakt::ids::ScopeId>>(parent_scope_id,[function_name, &result_ids, &result_scope, this](Jakt::types::ResolutionMixin mixin, JaktInternal::Optional<ByteString> name_override, bool is_alias) -> ErrorOr<typename Jakt::utility::IterationDecision<JaktInternal::Tuple<JaktInternal::DynamicArray<Jakt::ids::FunctionId>,Jakt::ids::ScopeId>>> {{
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 2736 | if (!mixin.mixin_functions){
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 2737 | return Jakt::utility::IterationDecision<JaktInternal::Tuple<JaktInternal::DynamicArray<Jakt::ids::FunctionId>,Jakt::ids::ScopeId>>::Continue();
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 2738 | }
      | ~
 2739 | NonnullRefPtr<Jakt::types::Scope> const scope = this->get_scope(mixin.scope_id);
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 2740 | JaktInternal::Optional<JaktInternal::DynamicArray<Jakt::ids::FunctionId>> const maybe_functions = scope->functions.get(function_name);
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 2741 | if (maybe_functions.has_value()){
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 2742 | result_ids.push_values(maybe_functions.value());
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 2743 | if (!result_scope.has_value()){
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 2744 | result_scope = mixin.scope_id;
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 2745 | }
      | ~
 2746 | }
      | ~
 2747 | return Jakt::utility::IterationDecision<JaktInternal::Tuple<JaktInternal::DynamicArray<Jakt::ids::FunctionId>,Jakt::ids::ScopeId>>::Continue();
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 2748 | }
      | ~
 2749 | }
      | ~
 2750 | ,ignore_mixin_scopes,root_scope)));
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/ljian/CTF/Kernel/serenity/Toolchain/Tarballs/jakt-1fed928d0abf08188e48fe765ab68e5047c05ec2/runtime/AK/Try.h:33:9: note: expanded from macro 'TRY'
   33 |         _temporary_result.release_value();                                                           \
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
[jakt/build] [239/240] Building CXX object CMakeFiles/jakt_stage1.dir/jakt_stage1_tmp/typechecker.cpp.o
FAILED: CMakeFiles/jakt_stage1.dir/jakt_stage1_tmp/typechecker.cpp.o 
/opt/homebrew/bin/ccache /usr/bin/c++  -I/Users/ljian/CTF/Kernel/serenity/Toolchain/Tarballs/jakt-1fed928d0abf08188e48fe765ab68e5047c05ec2/runtime -O3 -DNDEBUG -std=gnu++20 -arch arm64 -Wall -Wextra -Werror -Wno-maybe-uninitialized -Wno-unused-local-typedefs -Wno-unused-function -Wno-unused-variable -Wno-unused-parameter -Wno-unused-but-set-variable -Wno-unused-result -Wno-implicit-fallthrough -Wno-trigraphs -Wno-parentheses-equality -Wno-unqualified-std-cast-call -Wno-user-defined-literals -Wno-literal-suffix -Wno-return-type -Wno-deprecated-declarations -Wno-unknown-warning-option -Wno-unused-command-line-argument -Wno-unused-lambda-capture -Wno-reorder-ctor -Wno-dangling-reference -Wno-type-limits -Wno-unknown-attributes -Wno-invalid-offsetof -fno-exceptions -fdiagnostics-color=always -MD -MT CMakeFiles/jakt_stage1.dir/jakt_stage1_tmp/typechecker.cpp.o -MF CMakeFiles/jakt_stage1.dir/jakt_stage1_tmp/typechecker.cpp.o.d -o CMakeFiles/jakt_stage1.dir/jakt_stage1_tmp/typechecker.cpp.o -c /Users/ljian/CTF/Kernel/serenity/Toolchain/Build/jakt/jakt_stage1_tmp/typechecker.cpp
/Users/ljian/CTF/Kernel/serenity/Toolchain/Build/jakt/jakt_stage1_tmp/typechecker.cpp:9423:1: error: ignoring temporary created by a constructor declared with 'nodiscard' attribute [-Werror,-Wunused-value]
 9423 | TRY((this->typecheck_statement(var_decl_stmt,wrapped_scope_id,Jakt::types::SafetyMode::Safe(),JaktInternal::OptionalNone())));
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/ljian/CTF/Kernel/serenity/Toolchain/Tarballs/jakt-1fed928d0abf08188e48fe765ab68e5047c05ec2/runtime/AK/Try.h:33:9: note: expanded from macro 'TRY'
   33 |         _temporary_result.release_value();                                                           \
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
ninja: build stopped: subcommand failed.
build_jakt: /Users/ljian/CTF/Kernel/serenity/Toolchain/Local/jakt
[dependencies] Checking whether 'ninja' is available...
[dependencies] Checking whether 'cmake' is available...
[dependencies] Checking whether your C++ compiler works...
~/CTF/Kernel/serenity/Toolchain/Tarballs ~/CTF/Kernel/serenity/Toolchain
Using existing Jakt source directory
~/CTF/Kernel/serenity/Toolchain
Expected 1fed928d0abf08188e48fe765ab68e5047c05ec2 but found none (for local host)
~/CTF/Kernel/serenity/Toolchain/Build/jakt ~/CTF/Kernel/serenity/Toolchain
XXX configure jakt
[jakt/configure] -- The CXX compiler identification is AppleClang 17.0.0.17000013
[jakt/configure] -- The C compiler identification is AppleClang 17.0.0.17000013
[jakt/configure] -- Detecting CXX compiler ABI info
[jakt/configure] -- Detecting CXX compiler ABI info - done
[jakt/configure] -- Check for working CXX compiler: /usr/bin/c++ - skipped
[jakt/configure] -- Detecting CXX compile features
[jakt/configure] -- Detecting CXX compile features - done
[jakt/configure] -- Detecting C compiler ABI info
[jakt/configure] -- Detecting C compiler ABI info - done
[jakt/configure] -- Check for working C compiler: /usr/bin/cc - skipped
[jakt/configure] -- Detecting C compile features
[jakt/configure] -- Detecting C compile features - done
[jakt/configure] -- Performing Test CMAKE_HAVE_LIBC_PTHREAD
[jakt/configure] -- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
[jakt/configure] -- Found Threads: TRUE
[jakt/configure] -- Could NOT find Clang (missing: Clang_DIR)
[jakt/configure] -- Using none as the cpp import processor (using Clang=OFF, resource dir=)
[jakt/configure] -- Using preexisting AK directory from SERENITY_SOURCE_DIR: /Users/ljian/CTF/Kernel/serenity/Toolchain/../AK
[jakt/configure] CMake Warning (dev) at /opt/homebrew/share/cmake/Modules/FetchContent.cmake:1953 (message):
[jakt/configure]   Calling FetchContent_Populate(Serenity) is deprecated, call
[jakt/configure]   FetchContent_MakeAvailable(Serenity) instead.  Policy CMP0169 can be set to
[jakt/configure]   OLD to allow FetchContent_Populate(Serenity) to be called directly for now,
[jakt/configure]   but the ability to call it with declared details will be removed completely
[jakt/configure]   in a future version.
[jakt/configure] Call Stack (most recent call first):
[jakt/configure]   runtime/CMakeLists.txt:24 (FetchContent_Populate)
[jakt/configure] This warning is for project developers.  Use -Wno-dev to suppress it.
[jakt/configure] 
[jakt/configure] -- Making the jakt runtime available for arm64-unknown-darwin-unknown
[jakt/configure] -- If this is windows, '' should be nonempty
[jakt/configure] -- Building for target arm64-unknown-darwin-unknown
[jakt/configure] CMake Warning (dev) at cmake/symlink.cmake:41 (add_custom_command):
[jakt/configure]   The following keywords are not supported when using
[jakt/configure]   add_custom_command(TARGET): DEPENDS.
[jakt/configure] 
[jakt/configure]   Policy CMP0175 is not set: add_custom_command() rejects invalid arguments.
[jakt/configure]   Run "cmake --help-policy CMP0175" for policy details.  Use the cmake_policy
[jakt/configure]   command to set the policy and suppress this warning.
[jakt/configure] Call Stack (most recent call first):
[jakt/configure]   CMakeLists.txt:294 (add_symlink_command)
[jakt/configure] This warning is for project developers.  Use -Wno-dev to suppress it.
[jakt/configure] 
[jakt/configure] -- Configuring done (1.6s)
[jakt/configure] -- Generating done (0.0s)
[jakt/configure] -- Build files have been written to: /Users/ljian/CTF/Kernel/serenity/Toolchain/Build/jakt
XXX build jakt
[jakt/build] [0/2] Re-checking globbed directories...
[jakt/build] [238/240] Building CXX object CMakeFiles/jakt_stage1.dir/jakt_stage1_tmp/types.cpp.o
FAILED: CMakeFiles/jakt_stage1.dir/jakt_stage1_tmp/types.cpp.o 
/opt/homebrew/bin/ccache /usr/bin/c++  -I/Users/ljian/CTF/Kernel/serenity/Toolchain/Tarballs/jakt-1fed928d0abf08188e48fe765ab68e5047c05ec2/runtime -O3 -DNDEBUG -std=gnu++20 -arch arm64 -Wall -Wextra -Werror -Wno-maybe-uninitialized -Wno-unused-local-typedefs -Wno-unused-function -Wno-unused-variable -Wno-unused-parameter -Wno-unused-but-set-variable -Wno-unused-result -Wno-implicit-fallthrough -Wno-trigraphs -Wno-parentheses-equality -Wno-unqualified-std-cast-call -Wno-user-defined-literals -Wno-literal-suffix -Wno-return-type -Wno-deprecated-declarations -Wno-unknown-warning-option -Wno-unused-command-line-argument -Wno-unused-lambda-capture -Wno-reorder-ctor -Wno-dangling-reference -Wno-type-limits -Wno-unknown-attributes -Wno-invalid-offsetof -fno-exceptions -fdiagnostics-color=always -MD -MT CMakeFiles/jakt_stage1.dir/jakt_stage1_tmp/types.cpp.o -MF CMakeFiles/jakt_stage1.dir/jakt_stage1_tmp/types.cpp.o.d -o CMakeFiles/jakt_stage1.dir/jakt_stage1_tmp/types.cpp.o -c /Users/ljian/CTF/Kernel/serenity/Toolchain/Build/jakt/jakt_stage1_tmp/types.cpp
/Users/ljian/CTF/Kernel/serenity/Toolchain/Build/jakt/jakt_stage1_tmp/types.cpp:2735:1: error: ignoring temporary created by a constructor declared with 'nodiscard' attribute [-Werror,-Wunused-value]
 2735 | TRY((this->template for_each_scope_accessible_unqualified_from_scope<JaktInternal::Tuple<JaktInternal::DynamicArray<Jakt::ids::FunctionId>,Jakt::ids::ScopeId>>(parent_scope_id,[function_name, &result_ids, &result_scope, this](Jakt::types::ResolutionMixin mixin, JaktInternal::Optional<ByteString> name_override, bool is_alias) -> ErrorOr<typename Jakt::utility::IterationDecision<JaktInternal::Tuple<JaktInternal::DynamicArray<Jakt::ids::FunctionId>,Jakt::ids::ScopeId>>> {{
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 2736 | if (!mixin.mixin_functions){
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 2737 | return Jakt::utility::IterationDecision<JaktInternal::Tuple<JaktInternal::DynamicArray<Jakt::ids::FunctionId>,Jakt::ids::ScopeId>>::Continue();
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 2738 | }
      | ~
 2739 | NonnullRefPtr<Jakt::types::Scope> const scope = this->get_scope(mixin.scope_id);
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 2740 | JaktInternal::Optional<JaktInternal::DynamicArray<Jakt::ids::FunctionId>> const maybe_functions = scope->functions.get(function_name);
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 2741 | if (maybe_functions.has_value()){
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 2742 | result_ids.push_values(maybe_functions.value());
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 2743 | if (!result_scope.has_value()){
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 2744 | result_scope = mixin.scope_id;
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 2745 | }
      | ~
 2746 | }
      | ~
 2747 | return Jakt::utility::IterationDecision<JaktInternal::Tuple<JaktInternal::DynamicArray<Jakt::ids::FunctionId>,Jakt::ids::ScopeId>>::Continue();
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 2748 | }
      | ~
 2749 | }
      | ~
 2750 | ,ignore_mixin_scopes,root_scope)));
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/ljian/CTF/Kernel/serenity/Toolchain/Tarballs/jakt-1fed928d0abf08188e48fe765ab68e5047c05ec2/runtime/AK/Try.h:33:9: note: expanded from macro 'TRY'
   33 |         _temporary_result.release_value();                                                           \
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
[jakt/build] [239/240] Building CXX object CMakeFiles/jakt_stage1.dir/jakt_stage1_tmp/typechecker.cpp.o
FAILED: CMakeFiles/jakt_stage1.dir/jakt_stage1_tmp/typechecker.cpp.o 
/opt/homebrew/bin/ccache /usr/bin/c++  -I/Users/ljian/CTF/Kernel/serenity/Toolchain/Tarballs/jakt-1fed928d0abf08188e48fe765ab68e5047c05ec2/runtime -O3 -DNDEBUG -std=gnu++20 -arch arm64 -Wall -Wextra -Werror -Wno-maybe-uninitialized -Wno-unused-local-typedefs -Wno-unused-function -Wno-unused-variable -Wno-unused-parameter -Wno-unused-but-set-variable -Wno-unused-result -Wno-implicit-fallthrough -Wno-trigraphs -Wno-parentheses-equality -Wno-unqualified-std-cast-call -Wno-user-defined-literals -Wno-literal-suffix -Wno-return-type -Wno-deprecated-declarations -Wno-unknown-warning-option -Wno-unused-command-line-argument -Wno-unused-lambda-capture -Wno-reorder-ctor -Wno-dangling-reference -Wno-type-limits -Wno-unknown-attributes -Wno-invalid-offsetof -fno-exceptions -fdiagnostics-color=always -MD -MT CMakeFiles/jakt_stage1.dir/jakt_stage1_tmp/typechecker.cpp.o -MF CMakeFiles/jakt_stage1.dir/jakt_stage1_tmp/typechecker.cpp.o.d -o CMakeFiles/jakt_stage1.dir/jakt_stage1_tmp/typechecker.cpp.o -c /Users/ljian/CTF/Kernel/serenity/Toolchain/Build/jakt/jakt_stage1_tmp/typechecker.cpp
/Users/ljian/CTF/Kernel/serenity/Toolchain/Build/jakt/jakt_stage1_tmp/typechecker.cpp:9423:1: error: ignoring temporary created by a constructor declared with 'nodiscard' attribute [-Werror,-Wunused-value]
 9423 | TRY((this->typecheck_statement(var_decl_stmt,wrapped_scope_id,Jakt::types::SafetyMode::Safe(),JaktInternal::OptionalNone())));
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/ljian/CTF/Kernel/serenity/Toolchain/Tarballs/jakt-1fed928d0abf08188e48fe765ab68e5047c05ec2/runtime/AK/Try.h:33:9: note: expanded from macro 'TRY'
   33 |         _temporary_result.release_value();                                                           \
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
ninja: build stopped: subcommand failed.
[0/5] Performing build step for 'lagom'
ninja: no work to do.
[1/5] Performing install step for 'lagom'
[0/1] Install the project...
-- Install configuration: ""
[2/5] Performing build step for 'serenity'
[1/1] Running utility command for create_compat_ssp_libraries
[3/5] Performing install step for 'serenity'
[1/2] Install the project...
-- Install configuration: ""
[5/5] Completed 'serenity'
ninja: Entering directory `/Users/ljian/CTF/Kernel/serenity/Build/aarch64'
[1/2] Install the project...
-- Install configuration: ""
ninja: Entering directory `/Users/ljian/CTF/Kernel/serenity/Build/aarch64'
[0/1] cd /Users/ljian/CTF/Kernel/serenity/Build/aarch64 &&... /Users/ljian/CTF/Kernel/serenity/Meta/build-image-qemu.sh
Password:
checking existing image
e2fsck 1.47.2 (1-Jan-2025)
Pass 1: Checking inodes, blocks, and sizes
Pass 2: Checking directory structure
Pass 3: Checking directory connectivity
Pass 4: Checking reference counts
Pass 5: Checking group summary information
_disk_image: 35684/132608 files (0.2% non-contiguous), 249934/524288 blocks
done
mounting filesystem... mount_macfuse: mount point /Users/ljian/CTF/Kernel/serenity/Build/aarch64/mnt is itself on a macFUSE volume
mount failed but genext2fs exists, use it instead
installing base system... chmod: Unable to change file mode on mnt//res/fonts/CJKBiángRegular36.font: No such file or directory
chmod: Unable to change file mode on mnt//res/fonts/._CJKBiángRegular36.font: No such file or directory
rm: mnt: Resource busy
FAILED: CMakeFiles/qemu-image _disk_image /Users/ljian/CTF/Kernel/serenity/Build/aarch64/CMakeFiles/qemu-image /Users/ljian/CTF/Kernel/serenity/Build/aarch64/_disk_image 
cd /Users/ljian/CTF/Kernel/serenity/Build/aarch64 && /opt/homebrew/bin/cmake -E env SERENITY_SOURCE_DIR=/Users/ljian/CTF/Kernel/serenity SERENITY_ARCH=aarch64 SERENITY_TOOLCHAIN=GNU LLVM_VERSION=13.2.0 /Users/ljian/CTF/Kernel/serenity/Meta/build-image-qemu.sh
ninja: build stopped: subcommand failed.
```
My laptop is **mac M2**. 
And I don't know why ` Unable to change file mode on mnt//res/fonts/CJKBiángRegular36.font: No such file or directory`. 
How can we fix this?

--- a/Meta/build-image-qemu.sh
+++ b/Meta/build-image-qemu.sh
@@ -148,12 +148,12 @@ fi
 
 cleanup() {
     if [ -d mnt ]; then
-        if [ $use_genext2fs = 0 ] ; then
+        if [ $use_genext2fs = 0 ]; then
             printf "unmounting filesystem... "
-            if [ $USE_FUSE2FS -eq 1 ]; then
+            if [ $USE_FUSE2FS -eq 1 ] && command -v fusermount >/dev/null 2>&1; then
                 fusermount -u mnt || (sleep 1 && sync && fusermount -u mnt)
             else
-                umount mnt || ( sleep 1 && sync && umount mnt )
+                umount mnt || (sleep 1 && sync && umount mnt)
             fi
         fi
         rm -rf mnt

--- a/Meta/build-image-qemu.sh
+++ b/Meta/build-image-qemu.sh
@@ -169,7 +169,7 @@ cleanup() {
 trap cleanup EXIT
 
 script_path=$(cd -P -- "$(dirname -- "$0")" && pwd -P)
-"$script_path/build-root-filesystem.sh"
+SERENITY_USE_GENEXT2FS=$use_genext2fs "$script_path/build-root-filesystem.sh"
 
 if [ $use_genext2fs = 1 ]; then
     genext2fs -B 4096 -b $((DISK_SIZE_BYTES / 4096)) -i "${BYTES_PER_INODE}" -d mnt _disk_image || die "try increasing image size (genext2fs -b)"

--- a/Meta/build-root-filesystem.sh
+++ b/Meta/build-root-filesystem.sh
@@ -23,14 +23,30 @@ if ! command -v rsync >/dev/null; then
     die "Please install rsync."
 fi
 
+# On macOS, APFS returns filenames as NFC bytes, but fuse2fs via macFUSE only
+# accepts file creation with NFD-encoded names. Use --iconv=UTF-8,UTF-8-MAC to
+# convert sender filenames from NFC (UTF-8) to NFD (UTF-8-MAC) so that
+# non-ASCII filenames can be written to the ext2 disk image.
+# This is not needed when using genext2fs, which operates on a directory directly.
+RSYNC_ICONV=""
+if [ "${SERENITY_USE_GENEXT2FS:-0}" = "0" ] && [ "$(uname -s)" = "Darwin" ] && rsync --help 2>&1 | grep -q iconv; then
+    RSYNC_ICONV="--iconv=UTF-8,UTF-8-MAC"
+fi
+
 if rsync --chown 2>&1 | grep "missing argument" >/dev/null; then
-    rsync -aH --chown=0:0 --inplace --update "$SERENITY_SOURCE_DIR"/Base/ mnt/
-    rsync -aH --chown=0:0 --exclude="/usr/include" --inplace --update Root/ mnt/
-    rsync -aHL --chown=0:0 --inplace --update Root/usr/include/ mnt/usr/include/
+    # shellcheck disable=SC2086
+    rsync -aH $RSYNC_ICONV --chown=0:0 --inplace --update "$SERENITY_SOURCE_DIR"/Base/ mnt/
+    # shellcheck disable=SC2086
+    rsync -aH $RSYNC_ICONV --chown=0:0 --exclude="/usr/include" --inplace --update Root/ mnt/
+    # shellcheck disable=SC2086
+    rsync -aHL $RSYNC_ICONV --chown=0:0 --inplace --update Root/usr/include/ mnt/usr/include/
 else
-    rsync -aH --inplace --update "$SERENITY_SOURCE_DIR"/Base/ mnt/
-    rsync -aH --inplace --exclude="/usr/include" --update Root/ mnt/
-    rsync -aHL --inplace --update Root/usr/include/ mnt/usr/include/
+    # shellcheck disable=SC2086
+    rsync -aH $RSYNC_ICONV --inplace --update "$SERENITY_SOURCE_DIR"/Base/ mnt/
+    # shellcheck disable=SC2086
+    rsync -aH $RSYNC_ICONV --inplace --exclude="/usr/include" --update Root/ mnt/
+    # shellcheck disable=SC2086
+    rsync -aHL $RSYNC_ICONV --inplace --update Root/usr/include/ mnt/usr/include/
     chown -R 0:0 mnt/
 fi
 

--- a/Toolchain/BuildFuseExt2.sh
+++ b/Toolchain/BuildFuseExt2.sh
@@ -8,27 +8,75 @@ DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 
 exit_if_running_as_root "Do not run BuildFuseExt2.sh as root, parts of your Toolchain directory will become root-owned"
 
-export PATH="/usr/local/opt/m4/bin:$PATH"
-
 die() {
     echo "die: $*"
     exit 1
 }
 
-if [[ "$OSTYPE" != "darwin"* ]]; then
-    die "This script makes sense to be run only on macOS"
+SOURCE_DIR="$DIR/Tarballs/fuse-ext2"
+PATCH_DIR="$DIR/Patches/fuse-ext2/macos"
+
+EXTRA_CFLAGS=""
+EXTRA_LDFLAGS=""
+
+if [[ "$OSTYPE" == "darwin"* ]]; then
+    if command -v brew &>/dev/null; then
+        m4_prefix="$(brew --prefix m4 2>/dev/null)"
+        [ -n "$m4_prefix" ] && export PATH="${m4_prefix}/bin:$PATH"
+        E2FSPROGS_PREFIX="$(brew --prefix e2fsprogs 2>/dev/null)"
+        EXTRA_CFLAGS="-I${E2FSPROGS_PREFIX}/include"
+        EXTRA_LDFLAGS="-L${E2FSPROGS_PREFIX}/lib"
+    elif command -v pkg-config &>/dev/null && pkg-config --exists ext2fs; then
+        EXTRA_CFLAGS="$(pkg-config --cflags ext2fs)"
+        EXTRA_LDFLAGS="$(pkg-config --libs ext2fs)"
+    else
+        die "Could not find e2fsprogs. Install it via Homebrew (brew install e2fsprogs) or ensure pkg-config can find ext2fs."
+    fi
+    # Support older osxfuse installs that put headers in /usr/local/include/osxfuse
+    if [ -d "/usr/local/include/osxfuse" ]; then
+        EXTRA_CFLAGS="-I/usr/local/include/osxfuse ${EXTRA_CFLAGS}"
+    fi
+    MACFUSE_FRAMEWORK_DIR="/Library/Filesystems/macfuse.fs/Contents/Frameworks"
+    if [ -d "${MACFUSE_FRAMEWORK_DIR}/MFMount.framework" ]; then
+        EXTRA_LDFLAGS="${EXTRA_LDFLAGS} -F${MACFUSE_FRAMEWORK_DIR}"
+    fi
+    MACOS_PATCHES=true
+else
+    if command -v pkg-config &>/dev/null && pkg-config --exists ext2fs; then
+        EXTRA_CFLAGS="$(pkg-config --cflags ext2fs)"
+        EXTRA_LDFLAGS="$(pkg-config --libs ext2fs)"
+    else
+        EXTRA_LDFLAGS="-lext2fs"
+    fi
+    MACOS_PATCHES=false
 fi
 
 mkdir -p "$DIR"/Tarballs
 pushd "$DIR"/Tarballs
 
-if [ ! -d fuse-ext2 ]; then
-    git clone https://github.com/alperakcan/fuse-ext2.git	
+if [ -d "$SOURCE_DIR" ]; then
+    rm -rf "$SOURCE_DIR"
 fi
 
-cd fuse-ext2
+git clone --depth 1 https://github.com/alperakcan/fuse-ext2.git "$SOURCE_DIR"
+
+cd "$SOURCE_DIR"
+if [ "$MACOS_PATCHES" = true ]; then
+    for patch in "$PATCH_DIR"/*.patch; do
+        [ -f "$patch" ] || continue
+        patch -p1 < "$patch" > /dev/null
+    done
+fi
+
 ./autogen.sh
-CFLAGS="-I/usr/local/include/osxfuse/ -I/$(brew --prefix e2fsprogs)/include" LDFLAGS="-L$(brew --prefix e2fsprogs)/lib" ./configure
-make
+CFLAGS="${EXTRA_CFLAGS}" LDFLAGS="${EXTRA_LDFLAGS}" ./configure
+if [[ "$OSTYPE" == "darwin"* ]]; then
+    make \
+        "fuse_ext2_wait_LINK=\$(OBJC) \$(OBJCFLAGS) \$(AM_OBJCFLAGS) \$(AM_LDFLAGS) \$(LDFLAGS) \$(fuse_ext2_wait_LDFLAGS) -o \$@" \
+        "fuse_ext2_install_LINK=\$(OBJC) \$(OBJCFLAGS) \$(AM_OBJCFLAGS) \$(AM_LDFLAGS) \$(LDFLAGS) \$(fuse_ext2_install_LDFLAGS) -o \$@" \
+        "OBJCLINK=\$(OBJC) \$(OBJCFLAGS) \$(AM_OBJCFLAGS) \$(AM_LDFLAGS) \$(LDFLAGS) -o \$@"
+else
+    make
+fi
 sudo make install
 popd

--- a/Toolchain/Patches/fuse-ext2/macos/0001-Build-with-latest-macFUSE-on-macOS.patch
+++ b/Toolchain/Patches/fuse-ext2/macos/0001-Build-with-latest-macFUSE-on-macOS.patch
@@ -1,0 +1,56 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Bastiaan van der Plaat <bastiaan.v.d.plaat@gmail.com>
+Date: Tue, 5 May 2026 20:56:42 +0200
+Subject: [PATCH] Build with latest macFUSE on macOS
+
+Adjust the macOS build for current macFUSE headers and skip installing the
+legacy preference pane.
+
+Link: https://github.com/alperakcan/fuse-ext2/issues/149
+---
+ fuse-ext2/fuse-ext2.h    | 2 +-
+ fuse-ext2/op_getxattr.c  | 3 ++-
+ tools/macosx/Makefile.am | 2 --
+ 3 files changed, 3 insertions(+), 4 deletions(-)
+
+diff --git a/fuse-ext2/fuse-ext2.h b/fuse-ext2/fuse-ext2.h
+index 07fd1db..09399ed 100644
+--- a/fuse-ext2/fuse-ext2.h
++++ b/fuse-ext2/fuse-ext2.h
+@@ -178,7 +178,7 @@ int op_fgetattr (const char *path, struct stat *stbuf, struct fuse_file_info *fi
+ 
+ int op_getattr (const char *path, struct stat *stbuf);
+ 
+-int op_getxattr(const char *path, const char *name, char *value, size_t size);
++int op_getxattr(const char *path, const char *name, char *value, size_t size, uint32_t position);
+ 
+ ext2_file_t do_open (ext2_filsys e2fs, const char *path, int flags);
+ 
+diff --git a/fuse-ext2/op_getxattr.c b/fuse-ext2/op_getxattr.c
+index 27f95c3..eea5ae3 100644
+--- a/fuse-ext2/op_getxattr.c
++++ b/fuse-ext2/op_getxattr.c
+@@ -15,7 +15,8 @@
+ static int do_getxattr(ext2_filsys e2fs, struct ext2_inode *node, const char *name,
+ 		char *value, size_t size);
+ 
+-int op_getxattr(const char *path, const char *name, char *value, size_t size) {
++int op_getxattr(const char *path, const char *name, char *value, size_t size, uint32_t position) {
++	(void)position;
+ 	int rt;
+ 	ext2_ino_t ino;
+ 	struct ext2_inode inode;
+diff --git a/tools/macosx/Makefile.am b/tools/macosx/Makefile.am
+index 5afaaa4..533d09f 100644
+--- a/tools/macosx/Makefile.am
++++ b/tools/macosx/Makefile.am
+@@ -26,7 +26,5 @@ install-exec-local:
+ 	$(INSTALL) -m 644 $(top_srcdir)/tools/macosx/fuse-ext2.fs/Contents/Info.plist $(DESTDIR)/Library/Filesystems/fuse-ext2.fs/Contents/Info.plist
+ 	$(INSTALL) -m 644 $(top_srcdir)/tools/macosx/fuse-ext2.fs/Contents/PkgInfo $(DESTDIR)/Library/Filesystems/fuse-ext2.fs/Contents/PkgInfo
+ 	$(INSTALL) -m 644 $(top_srcdir)/tools/macosx/fuse-ext2.fs/Contents/Resources/English.lproj/InfoPlist.strings $(DESTDIR)/Library/Filesystems/fuse-ext2.fs/Contents/Resources/English.lproj/InfoPlist.strings
+-	$(INSTALL) -d $(DESTDIR)/Library/PreferencePanes
+-	cp -R $(top_srcdir)/tools/macosx/prefpane/build/Release/fuse-ext2.prefPane $(DESTDIR)/Library/PreferencePanes/fuse-ext2.prefPane
+ 
+ endif
+-- 
+2.49.0

--- a/Toolchain/Patches/fuse-ext2/macos/ReadMe.md
+++ b/Toolchain/Patches/fuse-ext2/macos/ReadMe.md
@@ -1,0 +1,14 @@
+# Patches for fuse-ext2 on macOS
+
+## `0001-Build-with-latest-macFUSE-on-macOS.patch`
+
+Build with latest macFUSE on macOS
+
+Adjusts the macOS build for current macFUSE headers and skips installing
+the legacy preference pane.
+
+The `op_getxattr` signature gains a `position` parameter required by the
+macFUSE 4.x FUSE API. The preference pane install step is removed because
+the prebuilt `.prefPane` bundle is no longer shipped in the source tree.
+
+https://github.com/alperakcan/fuse-ext2/issues/149

